src/test/scala/mesosphere/marathon/api/akkahttp/v2/TasksControllerTest.scala

```diff
@@ -0,0 +1,176 @@
+                         |  }, {
+                         |    "appId" : "app",
+                         |    "healthCheckResults" : [ ],
+                         |    "host" : "${instance2.hostname}",
+                         |    "id" : "${instance2.appTask.taskId.idString}",
+                         |    "ipAddresses" : [ ],
+                         |    "ports" : [ ],
+                         |    "servicePorts" : [ ],
+                         |    "slaveId" : "agent-1",
+                         |    "state" : "TASK_STAGING",
+                         |    "stagedAt" : "2015-04-09T12:30:00.000Z",
+                         |    "version" : "2015-04-09T12:30:00.000Z",
+                         |    "localVolumes" : [ ]
+                         |  } ]
+                         |}""".stripMargin
+        JsonTestHelper.assertThatJsonString(responseAs[String]).correspondsToJsonString(expected)
+      }
+    }
+
+    "list tasks as json with status filter" in new Fixture {
+      Given("some tasks in progress")
+      val clock = new SettableClock()
+      val app = AppDefinition(id = "/app".toPath, instances = 1)
+      val (runningInstance, runningTask) = runningInstanceAndItsTask(app, clock)
+      val (stagingInstance, stagingTask) = stagingInstanceAndItsTask(app, clock)
+
+      instanceTracker.instancesBySpec() returns Future.successful(InstanceTracker.InstancesBySpec.of(InstanceTracker.SpecInstances.forInstances("app".toPath, Seq(runningInstance, stagingInstance))))
+      groupManager.apps(any) returns Map(app.id -> Some(app))
+      healthCheckManager.statuses(any) returns Future.successful(Map.empty)
+
+      When("Getting the tasks list only with status staging")
+      Get(Uri./.withQuery(Query("status" -> "staging"))) ~> controller.route ~> check {
+        Then("Only one task should be serialized into response")
+        status should be(StatusCodes.OK)
+
+        responseAs[String] should include (stagingTask.taskId.idString)
+        responseAs[String] should not include (runningTask.taskId.idString)
+      }
+    }
+
+    "list tasks as json with multiple statuses" in new Fixture {
+      Given("some tasks in progress")
+      val clock = new SettableClock()
+      val app = AppDefinition(id = "/app".toPath, instances = 1)
+      val (runningInstance, runningTask) = runningInstanceAndItsTask(app, clock)
+      val (stagingInstance, stagingTask) = stagingInstanceAndItsTask(app, clock)
+
+      instanceTracker.instancesBySpec() returns Future.successful(InstanceTracker.InstancesBySpec.of(InstanceTracker.SpecInstances.forInstances("app".toPath, Seq(stagingInstance, runningInstance))))
+      groupManager.apps(any) returns Map(app.id -> Some(app))
+      healthCheckManager.statuses(any) returns Future.successful(Map.empty)
+
+      When("Getting the tasks list for running as well as staging")
+      Get(Uri./.withQuery(Query("status[]" -> "staging", "status[]" -> "running"))) ~> controller.route ~> check {
```

Comment:
Ah, the query parameter is called `status[]`? I don't think this is defined in the RAML.

Reply:
yeah, in the old tasks resource, there is status (single) but it also accepts array of status[] (the old php style, probably). I thought I should actually keep it backward compatible...

Reply:
I documented the status[] in RAML